src/components/Bookings/Bookings.jsx

```diff
@@ -1,10 +1,24 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import Search from "../Search/Search";
-import FakeBookings from "../../data/fakeBookings.json";
 import SearchResults from "../SearchResults/SearchResults";
 
 const Bookings = () => {
-  const [bookings, setBookings] = useState(FakeBookings);
+  const [bookings, setBookings] = useState([]);
+  const [loading, setLoading] = useState(true); 
+
+  useEffect(() => {
+    console.log("Component is mounted");
+    fetch("https://nw6-cyf-hotel.glitch.me/fakebookings")
+      .then(response => response.json())
+      .then(data => {
+        setBookings(data);
+        setLoading(false); 
+      })
+      .catch(error => {
+        console.error("Error fetching bookings:", error);
+        setLoading(false); 
```

Comment:
I am a bit confused here. Shouldt it be setloading(true) to prevent the component to be rendered? Because if the data is not fetched properly It would cause a problem in line 34. Component will still be rendered without proper data.

Reply:
Zeliha, If you look at line 7  **useState(true)**. The loading state is initially set to true . This means when the component first appears, it will show a loading.. msg. Later, when the data is fetched or there's an error, loading becomes false. Then you can render the component with the data or show an error message.